crates/exex/src/hints.rs

```diff
@@ -128,3 +156,119 @@ pub fn print_tx_hint() -> Hint {
         },
     )
 }
+
+/// Generates a hint to store block information in the `Environment` model.
+pub fn block_info_hint(block: SealedBlock, transaction: TransactionSignedEcRecovered) -> Hint {
```

Comment:
@ClementWalter Here, tell me if all the fields match what you expect. Since in the env structure, there is chain_id and sender, I concluded that this structure corresponds to both information of the block and transactions contained in this block. As a result, this hint is executed for each transaction of each block.

Reply:
I get your point but this is not going to hold, this is more a legacy of the fact that on stakrnet the context is only for one tx. I think that this it’s not so important rn, once the basic Cairo code is merged we can update

Reply:
Yes it seemed to me that it was necessary to do only one execution per block. So what we can do is merge as it is currently and then simplify, right?

But in this case it will be necessary to eliminate in the `env` struct all the fields specific to the transaction, right? For example the origin, chain id or gas price are elements specific to the transaction and not to the block. Here for example is where we can find the chain id in the transaction and not in the block

https://github.com/paradigmxyz/reth/blob/f37bd47e2ec46f19532c04214e8aeb8a74866ac9/crates/primitives/src/transaction/mod.rs#L189